opennms-webapp/src/main/webapp/account/selfService/passwordGate.jsp

```diff
@@ -0,0 +1,226 @@
+    padding: 7px;
+    padding-left: 28px;
+    padding-right: 28px;
+    font-size: 11px;
+    border-radius: 30px;
+    background-image: linear-gradient(to right, rgb(67, 194, 233), rgb(137, 230, 194));
+    border: none;
+  }
+
+  button:hover {
+    background-image: linear-gradient(to right, rgb(61, 168, 200), rgb(116, 187, 160));
+  }
+
+  .btn {
+    border-radius: 1.25rem;
+    color: #000;
+    font-size: 11px;
+  }
+
+  .btn-primary, .btn-secondary {
+    min-width: 80px;
+  }
+
+  .btn-secondary {
+    margin-left: 4px;
+    background-image: linear-gradient(to right, rgb(84, 142, 131), rgb(104, 156, 139));
+  }
+
+  .horizon {
+    margin-left: 20%;
+  }
+
+  label.pg-text, span.pg-text {
+    color: #fff;
+  }
+</style>
+
+<%
+  // Spring Security remembers the requested URL and redirects after a successful login.
+  // If the session_expired parameter is set, it is known that this is a Javascript redirect.
+  // In this case, we remove the remembered request attribute so the user is forwarded to the login page instead.
+  if (request.getParameter("session_expired") != null
+    && request.getParameter("session_expired").equals("true")
+    && session != null) {
+    session.removeAttribute("SPRING_SECURITY_SAVED_REQUEST");
+  }
+%>
+<script type="text/javascript">
+  function verifyGoForm(event) {
+    if (event.submitter.id === 'btn_skip') {
+        return true;
+    }
+
+    if (document.goForm.pass1.value == document.goForm.pass2.value) {
+      let newPassword = document.goForm.pass1.value
+      const passwordRegex = /${fn:escapeXml(PasswordGateActionServlet.PASSWORD_REGEX)}/;
+      const sameCharacterRegex = /${fn:escapeXml(PasswordGateActionServlet.SAME_CHARACTER_REGEX)}/;
+
+      if (newPassword.match(passwordRegex) && !newPassword.match(sameCharacterRegex)) {
+        document.goForm.currentPassword.value = document.goForm.oldpass.value;
+        document.goForm.newPassword.value = document.goForm.pass1.value;
+        return true;
+      } else {
+        alert("Password complexity is not correct! Please use at least 12 characters, consisting of 1 special character, 1 upper case letter, 1 lower case letter and 1 number. Identical strings with 6 or more characters in a row are also not allowed.");
+        return false;
+      }
+    } else {
+      alert("The two new password fields do not match!");
+      return false;
+    }
+  }
+
+  window.onload = function() {
+    var oldPass = document.getElementById("loginForm:input_oldpass") || document.getElementById("input_oldpass");
+    var pass1 = document.getElementById("loginForm:input_pass1") || document.getElementById("input_pass1");
+    var pass2 = document.getElementById("loginForm:input_pass2") || document.getElementById("input_pass2");
+
+    oldPass.value = '';
+    pass1.value = '';
+    pass2.value = '';
+  }
+</script>
+
+<div class="login-page">
+    <div class="card login-form rounded">
+        <div style="padding-bottom: 36px; padding-top: 60px">
```

Comment:
Super minor nit, but you're mixing inline styles and classes here, and also in the footer. Not a huge deal normally for a small one off but this looks like a new file and there's so much CSS just above this. :)

Reply:
Thanks for taking a look! Will wait til Ben or Christian review and approve due to the Java and server-side stuff, but wanted a FE dev to take a look as well :)

I will update this to use a CSS class here and possibly `login.jsp` in a subsequent PR.